deploy/1-deploy-oracles.ts

```diff
@@ -29,18 +29,27 @@ const func: DeployFunction = async function ({ getNamedAccounts, deployments, ne
   const accessControlManagerAddress = network.live ? ADDRESSES[networkName].acm : accessControlManager?.address;
   const proxyOwnerAddress = network.live ? ADDRESSES[networkName].timelock : deployer;
 
+  const defaultProxyAdmin = await hre.artifacts.readArtifact(
+    "hardhat-deploy/solc_0.8/openzeppelin/proxy/transparent/ProxyAdmin.sol:ProxyAdmin",
+  );
+
   await deploy("BoundValidator", {
     from: deployer,
     log: true,
     deterministicDeployment: false,
+    skipIfAlreadyDeployed: true,
     args: [],
     proxy: {
       owner: proxyOwnerAddress,
-      proxyContract: "OptimizedTransparentProxy",
+      proxyContract: "OptimizedTransparentUpgradeableProxy",
```

Comment:
Any mandatory reason to make this change?

Reply:
`OptimizedTransparentProxy` is basically a flag. Hardhat-deploy plugin checks this and adds the `OptimizedTransparentUpgradeableProxy` and the `DefaultProxyAdmin` fields in the places I have added them in. I only have brought it one level up from the plugin in the node modules. The reason for this is that the contracts `OptimizedTransparentUpgradeableProxy` and `DefaultProxyAdmin` that the hardhat-deploy plugin fetches from the artifact is not zk compatible causing the deployments to fail. So we bought it one level up to our repo, added them to compile using zksync compiler.